enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,545 @@
+```
+
+The PO manager will be responsible for acting as a proxy between the CVO, and the underlying PO packages during phase 0. This can be done by reconciling an "aggregate" ClusterOperator that reflects the overall state of installing PO packages. Future phases may require PO packages to reconcile OLM's OperatorCondition API, or the directly manage a ClusterOperator resource to express their health and overall state.
+
+> Note: The plain provisioner has the current limitation around applying Bundle contents that will produce runtime issues. In the case that unpacked Bundle contents contain a Deployment manifest, and that manifest references an invalid container image, then the result would be an unavailable container process after being applied to the cluster. Despite the presence of runtime issues here, the BundleDeployment that manages those manifests would reflect a successful installation state. Addressing this limitation is out-of-scope for phase 0, and will be addressed in future phases.
+
+When reconciling a PO custom resource, the PO manager will need to propagate the underlying BundleDeployment's status back to the PO custom resource that's currently being processed, and relaying that individual PO status information back to the "aggregate" ClusterOperator that controller is managing.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+CVO->>PO custom resource: Generates
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk. These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+When the "TechPreviewNoUpgrades" feature set is enabled for a cluster, several API extensions will be added during phase 0.
+
+#### OpenShift API(s)
+
+> Note: The following API(s) will live in the openshift/api repository.
+
+- Introduces a top-level PlatformOperator custom-scoped v1alpha1 CRD:
+
+```go
+const (
+    TypeSourced = "Sourced"
+    TypeApplied = "Applied"
+
+    ReasonSourceFailed     = "SourceFailed"
+    ReasonSourceSuccessful = "SourceSuccessful"
+    ReasonApplyFailed      = "ApplyFailed"
+    ReasonApplySuccessful  = "ApplySuccessful"
+)
```

Comment:
These will want comments explaining what they are for, to me they look like Condition types and reasons? You can probably keep those just in your operator repo rather than in the API unless there's some specific reason you intend to share these?

Reply:
Nope, no explicit reason here. I'm cool with moving these in-tree to our repository. I'll update that in the next round of commits.